src/test/java/bio/terra/service/duos/DuosClientTest.java

```diff
@@ -106,9 +147,85 @@ public void testGetDatasetThrowsWhenIdMalformed() {
         equalTo(expectedEx));
   }
 
+  @Test
+  public void testGetApprovedUsers() {
+    var expectedBody =
+        new DuosDatasetApprovedUsers(
+            List.of("a", "b", "c").stream().map(DuosDatasetApprovedUser::new).toList());
+    when(restTemplate.exchange(
+            eq(duosClient.getApprovedUsersUrl(DUOS_ID)),
+            eq(HttpMethod.GET),
+            argThat(this::hasTdrSaToken),
+            eq(DuosDatasetApprovedUsers.class)))
+        .thenReturn(ResponseEntity.ok(expectedBody));
+
+    assertThat(
+        "Authorized users are returned",
+        duosClient.getApprovedUsers(DUOS_ID),
+        equalTo(expectedBody));
+  }
+
+  @Test
+  public void testGetApprovedUsersThrows() {
+    var expectedEx = new HttpClientErrorException(HttpStatus.NOT_FOUND);
+    when(restTemplate.exchange(
+            eq(duosClient.getApprovedUsersUrl(DUOS_ID)),
+            eq(HttpMethod.GET),
+            argThat(this::hasTdrSaToken),
+            eq(DuosDatasetApprovedUsers.class)))
+        .thenThrow(expectedEx);
+
+    DuosDatasetNotFoundException thrown =
+        assertThrows(
+            DuosDatasetNotFoundException.class, () -> duosClient.getApprovedUsers(DUOS_ID));
+    assertThat(
+        "Exception thrown when DUOS dataset not found for the supplied ID",
+        thrown.getCause(),
+        equalTo(expectedEx));
+  }
+
+  @Test
+  public void testConvertDuosExToDataRepoEx() {
+    var badRequestEx = new HttpClientErrorException(HttpStatus.BAD_REQUEST);
+    assertThat(
+        DuosClient.convertDuosExToDataRepoEx(badRequestEx, DUOS_ID),
+        instanceOf(DuosDatasetBadRequestException.class));
+
+    var notFoundEx = new HttpClientErrorException(HttpStatus.NOT_FOUND);
+    assertThat(
+        DuosClient.convertDuosExToDataRepoEx(notFoundEx, DUOS_ID),
+        instanceOf(DuosDatasetNotFoundException.class));
+
+    var unexpectedEx = new HttpClientErrorException(HttpStatus.I_AM_A_TEAPOT);
+    assertThat(
+        DuosClient.convertDuosExToDataRepoEx(unexpectedEx, DUOS_ID),
+        instanceOf(DuosInternalServerErrorException.class));
+
+    var internalServerErrorEx = new HttpClientErrorException(HttpStatus.INTERNAL_SERVER_ERROR);
+    assertThat(
+        DuosClient.convertDuosExToDataRepoEx(internalServerErrorEx, DUOS_ID),
+        instanceOf(DuosInternalServerErrorException.class));
+  }
+
+  // Utilities for examining HttpEntity parameters
+  private List<String> getAuthorizations(HttpEntity httpEntity) {
```

Comment:
`HttpEntity` is a generic type and it's generally considered bad form to use a generic type without a type argument, and the java compiler will generate a warning (as does IntelliJ). It's usually easy to fix by either using a wildcard type (`<?>`) or by making the class or method that uses it generic. In this case you can write
```suggestion
  private <T> List<String> getAuthorizations(HttpEntity<T> httpEntity) {
```

Note that these helper methods could all be made `static`. If you do that, you'll need to change the method references from `this::isUnauthenticated` to `DuosClientTest::isUnauthenticated`. My thinking is that making a method static tells the reader that the method doesn't use any class state. In a test class it's less important but IMO it's still a nice rule to follow.

Reply:
Ahh, I missed those warnings, thanks for pointing them out.  I agree with you re: what's implied by making these static methods, and made those changes as well.